CMakeLists.txt

```diff
@@ -212,16 +210,9 @@ if(WINDOWS)
     )
 endif()
 if(UMF_PROXY_LIB_BASED_ON_POOL STREQUAL SCALABLE)
-    if(UMF_BUILD_LIBUMF_POOL_SCALABLE)
-        set(UMF_PROXY_LIB_ENABLED ON)
-        set(PROXY_LIB_USES_SCALABLE_POOL ON)
-        set(PROXY_LIBS umf scalable_pool)
-    else()
-        message(
-            STATUS
-                "Disabling the proxy library, because UMF_PROXY_LIB_BASED_ON_POOL==SCALABLE but UMF_BUILD_LIBUMF_POOL_SCALABLE is OFF"
-        )
-    endif()
+    set(UMF_PROXY_LIB_ENABLED ON)
```

Comment:
There should be a check if the TBB was found by CMake. Such a check is also needed for an example using the Scalable Pool and benchmarks as now they use TBB unconditionally. I think that the build of these binaries will fail in case no TBB is found.

Reply:
+1
tests too should be conditional based on ttb avaibility

Reply:
I disagree - we could build tests and examples always because TBB is not a build-time dependency. I agree that tests and examples that use Scalable Pool require TBB installed on the system but we wouldn't ever run them without needed dependencies in our CI - similar to GPU-related tests.

Reply:
I fully agree with @bratpiorka

Reply:
TESTS should not fail if there is no bug in the code. This is why people created tools like automake, cmake.. to manage dependencies, and solve this kind of problems. 

Assumption that we have something configured on CI, do not free us from responsibility to provide stable development environment for ourself and external contributors. Even we have in PR template "All tests pass locally" checkmark. How developer can ensure it - if some tests fails on master as they do not have something installed on the platform, platform is misconfigured, did not selected correct options in cmake, or they do not have hardware which was not available for them. Do they really need to remember which tests fails, and check if more tests fails after their changes.

In opensource test are part of the software - some good open source project even encourages users to run tests if they build from source. Package maintainers use tests to check if package is build correctly on this system. It is important to provide stable, and easy to run tests.

IN case of examples it is more then critical - user has to get meaningful information about missing dependency during build. But now it will get random error from the library while running example about dlopen while running an example. 

How we should resolve this issue - 
Examples should not build if there is no TBB on system. User should get a meaningful error (or warning) during cmake command about this issue. 
Test should SKIP if TBB is not available. 
Also you should create an issue (or PR) to introduce --fail-on-skip in tests and use this option in CI to ensure that we don't have silent skips, while we are expecting tests to run.

Reply:
as we talked in the morning, I guess I agree with Łukasz.

In examples, we should definetely catch this in CMake.

In tests, we could introduce a separate flag (as you proposed, similarly to `UMF_BUILD_GPU_TESTS`) to enable these tests, otherwise print a CMake warning and omit adding these tests perhaps...

Reply:
> In examples, we should definetely catch this in CMake.

It should be done in the `examples/CMakeLists.txt`

Reply:
> > In examples, we should definetely catch this in CMake.
> 
> what if tbb library is not in the default search path? Why should we detect TBB availability in the build time, not runtime?

This is a good call - then we should detect it in example and return meaningful error about this issue